src/main/java/ecostruxure/rate/calculator/bll/service/CurrencyService.java

```diff
@@ -30,6 +30,13 @@ public Currency get(String code) throws Exception {
         return currencyDAO.get(code);
     }
 
+    public void importCurrencies(Currency[] currencies) throws Exception {
+        this.currencyDAO.removeAllCurrencies();
```

Comment:
Please use the ON CONFLICT function in Postgres instead of doing to seperate SQL calls. This should be added in the query in DAO, around line 100. Example is in ProjectDAO line 276-278